### PR TITLE
Add pulp-ui to pulp image

### LINKED ISSUE #664: Add Pulp UI to All-In-One image
We're working on a UI!! [0][1] We want this UI to present by default in our images, so first step is to add it to the single container image. 

[0] https://discourse.pulpproject.org/t/pulp-ui-working-group-forming/1704/
[1] https://github.com/pulp/pulp-ui

### PR DESCRIPTION
**Pull request by gerrod3**
There's many different ways to serve these files, here's one way I chose. Would like some feedback on this approach.

TODO:

- [x] Add logic in CI to find ui url
- [ ] Add support for nightly image??
- [x] Add support for templating a config file for values like API_ROOT (maybe in a future PR once we decide how we will do this in the ui)
- [x] Add issue changelog

### CODE REVIEW

File: images/pulp/stable/Containerfile [hunk 1]
```diff
@@ -39,3 +41,10 @@ USER root:root
 RUN ln $(pip3 show pulp_ansible | sed -n -e 's/Location: //p')/pulp_ansible/app/webserver_snippets/nginx.conf /etc/nginx/pulp/pulp_ansible.conf
 RUN ln $(pip3 show pulp_container | sed -n -e 's/Location: //p')/pulp_container/app/webserver_snippets/nginx.conf /etc/nginx/pulp/pulp_container.conf
 RUN ln $(pip3 show pulp_python | sed -n -e 's/Location: //p')/pulp_python/app/webserver_snippets/nginx.conf /etc/nginx/pulp/pulp_python.conf
+
+RUN <<EOF
+if [[ "$PULP_UI_URL" ]]; then
+    mkdir -p /usr/share/nginx/static/pulp_ui
```
Comment (mdellweg) [inline images/pulp/stable/Containerfile]: Is this the new /var/www ?
It's OK...

File: images/pulp/stable/Containerfile [hunk 2]
```diff
@@ -12,7 +12,9 @@ ARG PULP_MAVEN_VERSION=""
 ARG PULP_PYTHON_VERSION=""
 ARG PULP_RPM_VERSION=""
 ARG PULP_OSTREE_VERSION=""
+ARG PULP_UI_URL=""
 
+ENV PULP_UI=true
```
Comment (mdellweg) [inline images/pulp/stable/Containerfile]: This should probably depend on PULP_UI_URL... Or let's just make PULP_UI_URL the env var?

File: images/s6_assets/nginx.conf.j2 [hunk 3]
```diff
@@ -105,6 +105,17 @@ http {
 
         include pulp/*.conf;
 
+        {% if ui | default(false) -%}
```
Comment (mdellweg) [inline images/s6_assets/nginx.conf.j2:108]: This could be `if ui_url != ""`?

**Review by mdellweg — COMMENTED** [review]
I like it a lot.

File: images/pulp/stable/Containerfile (hunk 1 above)
Comment (gerrod3) [reply images/pulp/stable/Containerfile]: Where should these files live? I was thinking I could put them in the pulp static directory (default: /var/lib/operator/static/), but I was worried if the user ran collectstatic with --clean then they would lose their UI as well.

File: images/pulp/stable/Containerfile [hunk 4]
```diff
@@ -39,3 +41,10 @@ USER root:root
 RUN ln $(pip3 show pulp_ansible | sed -n -e 's/Location: //p')/pulp_ansible/app/webserver_snippets/nginx.conf /etc/nginx/pulp/pulp_ansible.conf
 RUN ln $(pip3 show pulp_container | sed -n -e 's/Location: //p')/pulp_container/app/webserver_snippets/nginx.conf /etc/nginx/pulp/pulp_container.conf
 RUN ln $(pip3 show pulp_python | sed -n -e 's/Location: //p')/pulp_python/app/webserver_snippets/nginx.conf /etc/nginx/pulp/pulp_python.conf
+
+RUN <<EOF
+  if [[ "$PULP_UI_URL" ]]; then
+    mkdir -p "${PULP_STATIC_ROOT}pulp_ui"
+    curl -Ls $PULP_UI_URL | tar -xzv -C "${PULP_STATIC_ROOT}pulp_ui"
+  fi
+EOF
```
Comment (mdellweg) [inline images/pulp/stable/Containerfile]: I haven't seen this "here document" thing in a container file before.
Can you try with `\` on each lines end instead?

Comment (gerrod3) [reply images/pulp/stable/Containerfile]: I couldn't get it to work with `\`.

Comment (gerrod3) [reply images/pulp/stable/Containerfile]: Ok I got a version with `\` to work, had to switch to using single `[` for the if check.

**Review by ggainey — APPROVED** [review]
This all looks reasonable now, nice work.

**Review by mdellweg — APPROVED** [review]

**Pull request merged**